Bug squishing and a new toggle

bigbeau opened the pull request:
The "nth-child" had the wrong ID, off by 1, so the toggles were hiding the wrong things and missing the new and trending tab. I fixed that, and then added a new toggle for hiding the "X games from wishlist are now on sale" tab for those that might want it.

Thanks for making this!!

Edit: sorry about the messy commits, it was 3 am when I did this

bigbeau committed:
Add files via upload

Fix wrong child number + add ability to hide wishlist

bigbeau committed:
Add files via upload

bigbeau committed:
Add files via upload

fix inject and add credit for myself :)

bigbeau committed:
today I learned how to spell

bigbeau committed:
move some stuff

bigbeau committed:
haha im dumb

if only I could read

bigbeau committed:
fix numbers and add new toggle

bigbeau committed:
add new toggle for wishlist hiding

Also made the desc not silly this time :P

bigbeau committed:
Delete Home_concealer/theme.json

bigbeau committed:
add new toggle for wishlist hiding

also had to delete and reup because I wanted to make the desc more serious and not silly this time cuz professionalism or something

RaZeSloth commented:
Thank you for the pr! Will merge if it lgtm.

RaZeSloth commented:
Interesting, the game carousels on the recommended tab seem to be broken... might be a client thing? Are you running the some different steam client instead of the stable version? I remember having something similar when I was on the beta client.

bigbeau commented:
Are you referring to the "Something went wrong while displaying this content" errors? I got those before my changes. I'm running OS Beta but Client Stable (Decky breaks often when using Client Beta channels)

`Error reference: "Shared SteamUI_8237455_d73444412de07a9c 
Cannot read property ' recomputeGridSize' of Undefined`

The carousel that fails to load for me is always the Play Next one, which is not one I touched. Seems to do it whenever it's hidden. I couldn't get anything else to cause one, only the Play Next. Happens pretty reliably from sleep, restart, whenever that carousel is set to hidden.

Other than that, carousels work fine for me, I can scroll through them just the same as before. It just seems to have an issue loading that one.

bigbeau commented:
Ok so I dug around a little bit...

[recomputeGridSize is, I think, a react-virtualized thing](https://github.com/bvaughn/react-virtualized/blob/master/docs/Grid.md#recomputegridsize--columnindex-number-rowindex-number-). I have no experience in anything like that so I wouldn't even know where to begin with that... if it's something that can even be fixed.

What I do know is it calls itself ErrorBoundary and just adding a `display: none` to it in the styles tab hides it, like so:
![image](https://github.com/RaZeSloth/DeckyThemes/assets/15239968/5ffb1200-f84c-4236-b10a-95685dbaf241)

it doesn't seem to cause any issues, probably only exists so the carousel doesn't load in with a bunch of missing images. Credit to this theory is that it spams console with failed resource loading attempts(all of them images) that all start with "steamloopback.host" which is a [valve domain](https://who.is/dns/steamloopback.host) that, apparently, nobody knows what it does. This only happens when that specific carousel is hidden so I don't know what's so special about it, maybe because it pulls from your library rather than the store? Idunno. 

Is it good to hide errors? well no probably not but if you consider that if someone wanted to hide that carousel anyway, they probably wouldn't care if it were unable to load it's content :P

bigbeau committed:
Attempt to "fix" the weird error

The error (which I've determined to be ok to ignore until proven otherwise) is now hidden along with the carousel. It now achieves the desired effect.

bigbeau commented:
The bandaid works for me so let me know if it works (or doesn't) for you :)

RaZeSloth commented:
Pretty sure it worked, I am not getting the error. But I am pretty sure the "nth-child" increase by one off issue is due to you having the "games from wishlist are now on sale" block. Which makes it off without the "games from wishlist are now on sale" block. What do you think is the way to go about this?

bigbeau commented:
Do you not have this block?
![image](https://github.com/RaZeSloth/DeckyThemes/assets/15239968/5cc14f3b-b9da-4975-a688-c490609afa67)

While I was writing this I decided to test if this block appearing caused any issues, and yeah, it does. As you suggested, when the block is present, it shifts the nth-child numbers of all subsequent blocks by 1, inserting itself at 3 in the list, and thus pushing "new and trending" from 5 to 6 (which in the original plugin couldn't be hidden because the hide switch assumed it was number 5). I assume when you made the plugin, you didn't have this block, so it didn't affect you.

To test this, I went though my wishlist and removed the games currently on sale, and lo and behold, my "fix" for the nth-child bug started to cause the same issue (only this time, while the ID numbers were shifted, I was still able to hide the block without leaving  the last block, "new and trending", because it's ID had moved from 6 back to 5)

The only way to fix this would be to not use nth-child as a means of identifying which block to hide.

bigbeau commented:
I'm looking now and I don't really see a way to identify them outside of just figuring out which child it is. Maybe their contents like the header can be used as a means of identifying them but I just don't know. I haven't fully read up on what the capabilities of CSS Loader and whether or not that can be done how we want. Outside of the header, they're basically identical except for properties like innerText, innerHTML, outerText... again, not even sure if CSS Loader can use that but I'll look into it
